DXMainClient/Online/Connection.cs

```diff
@@ -324,6 +326,110 @@ private void HandleComm(object client)
+
+                foreach (var server in Servers)
+                {
+                    var serverHostnameOrIPAddress = server.Host;
+                    var serverName = server.Name;
+                    var serverPorts = server.Ports;
+
+                    var dnsAndPingTask = new Task(() =>
+                    {
+                        Logger.Log($"Attempting to DNS resolve {serverName} ({serverHostnameOrIPAddress}).");
+
+                        // If hostNameOrAddress is an IP address, this address is returned without querying the DNS server.
+                        var serverIPAddresses = Dns.GetHostAddresses(serverHostnameOrIPAddress).Where(item => item.AddressFamily == AddressFamily.InterNetwork).ToList();
+
+                        Logger.Log($"DNS resolved {serverName} ({serverHostnameOrIPAddress}): {string.Join(", ", serverIPAddresses.Select(item => item.ToString()))}");
+
+                        var pingTasks = new List<Task>();
+
+                        foreach (var serverIPAddress in serverIPAddresses)
+                        {
+                            var theIPAddress = serverIPAddress;
+
+                            if (availableServerAndLatencyDict.Any(item => item.Key.Host == theIPAddress.ToString()))
+                            {
+                                Logger.Log($"Skipped a duplicate IP from {serverName} ({theIPAddress}).");
+
+                                continue;
+                            }
+
+                            if (failedServerIPs.Contains(theIPAddress.ToString()))
+                            {
+                                Logger.Log($"Skipped a failed server {serverName} ({theIPAddress}).");
+
+                                continue;
+                            }
+
+                            var pingTask = new Task(() =>
+                            {
+                                Logger.Log($"Attempting to ping {serverName} ({theIPAddress}).");
+
+                                var pingReply = new Ping().Send(theIPAddress, MAXIMUM_LATENCY);
+
+                                if (pingReply.Status == IPStatus.Success)
+                                {
+                                    var pingInMs = pingReply.RoundtripTime;
+
+                                    Logger.Log($"The latency in milliseconds to the server {serverName} ({theIPAddress}): {pingInMs}.");
+
+                                    availableServerAndLatencyDict.Add(new Server(theIPAddress.ToString(), serverName, serverPorts), pingInMs);
+                                }
+                                else
+                                {
+                                    if (pingReply.Status == IPStatus.TimedOut)
+                                    {
+                                        Logger.Log($"Pinging the server {serverName} ({theIPAddress}) timed out!");
+                                    }
+                                }
```

Comment:
Those 4 braces can be omitted. You should use `else if` here and write the logger call in a single line without braces.

Reply:
Fixed in c448a0473f5abb25e469f6ee78691bace0932b62 .